File: backend/app/schemas/site.py
"""
Site Schema
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class SiteBase(BaseModel):
    """站点基础模型"""

    name: str = Field(..., description="站点名称")
    domain: str | None = Field(None, description="站点域名")
    url: str = Field(..., description="站点URL")
    cookie: str | None = Field(None, description="Cookie")
    passkey: str | None = Field(None, description="Passkey")
    username: str | None = Field(None, description="用户名")
    password: str | None = Field(None, description="密码")
    proxy: str | None = Field(None, description="代理")
    ua: str | None = Field(None, description="User-Agent")
    timeout: int = Field(30, description="超时时间（秒）")
    rss: str | None = Field(None, description="RSS地址")
    rss_interval: int = Field(60, description="RSS刷新间隔（秒）")
    downloader: str = Field("qbittorrent", description="下载器类型")
    priority: int = Field(1, description="优先级")
    enabled: bool = Field(True, description="是否启用")
    site_type: str = Field("resource", description="站点类型")


class SiteCreate(SiteBase):
    """创建站点"""

    pass


class SiteUpdate(BaseModel):
    """更新站点"""

    name: str | None = None
    domain: str | None = None
    url: str | None = None
    cookie: str | None = None
    passkey: str | None = None
    username: str | None = None
    password: str | None = None
    proxy: str | None = None
    ua: str | None = None
    timeout: int | None = None
    rss: str | None = None
    rss_interval: int | None = None
    downloader: str | None = None
    priority: int | None = None
    enabled: bool | None = None
    site_type: str | None = None


class SiteResponse(SiteBase):
    """站点响应模型"""

    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SiteListResponse(BaseModel):
    """站点列表响应"""

    total: int
    items: list[SiteResponse]


class TestSiteRequest(BaseModel):
    """测试站点连接请求"""

    id: int = Field(..., description="站点ID")


class TestSiteResponse(BaseModel):
    """测试站点连接响应"""

    success: bool
    message: str

File: frontend/src/api/site.ts
import request from './request'

export interface Site {
  id: number
  name: string
  domain?: string
  url: string
  cookie?: string
  passkey?: string
  username?: string
  password?: string
  proxy?: string
  ua?: string
  timeout: number
  rss?: string
  rss_interval: number
  downloader: string
  priority: number
  enabled: boolean
  site_type: string
  created_at: string
  updated_at: string
}

export interface SiteCreate {
  name: string
  domain?: string
  url: string
  cookie?: string
  passkey?: string
  username?: string
  password?: string
  proxy?: string
  ua?: string
  timeout?: number
  rss?: string
  rss_interval?: number
  downloader?: string
  priority?: number
  enabled?: boolean
  site_type?: string
}

export interface SiteUpdate {
  name?: string
  domain?: string
  url?: string
  cookie?: string
  passkey?: string
  username?: string
  password?: string
  proxy?: string
  ua?: string
  timeout?: number
  rss?: string
  rss_interval?: number
  downloader?: string
  priority?: number
  enabled?: boolean
  site_type?: string
}

export interface SiteListResponse {
  total: number
  items: Site[]
}

export interface TestSiteRequest {
  id: number
}

export interface TestSiteResponse {
  success: boolean
  message: string
}

export const siteApi = {
  /**
   * 获取站点列表
   */
  getSites: (params?: {
    skip?: number
    limit?: number
    enabled?: boolean
    downloader?: string
  }) => {
    return request.get<SiteListResponse>('/api/v1/sites', { params })
  },

  /**
   * 获取启用的站点
   */
  getEnabledSites: () => {
    return request.get<Site[]>('/api/v1/sites/enabled')
  },

  /**
   * 获取站点详情
   */
  getSite: (id: number) => {
    return request.get<Site>(`/api/v1/sites/${id}`)
  },

  /**
   * 创建站点
   */
  createSite: (data: SiteCreate) => {
    return request.post<Site>('/api/v1/sites', data)
  },

  /**
   * 更新站点
   */
  updateSite: (id: number, data: SiteUpdate) => {
    return request.put<Site>(`/api/v1/sites/${id}`, data)
  },

  /**
   * 删除站点
   */
  deleteSite: (id: number) => {
    return request.delete(`/api/v1/sites/${id}`)
  },

  /**
   * 切换站点启用状态
   */
  toggleSite: (id: number) => {
    return request.post<Site>(`/api/v1/sites/${id}/toggle`)
  },

  /**
   * 测试站点连接
   */
  testSite: (data: TestSiteRequest) => {
    return request.post<TestSiteResponse>('/api/v1/sites/test', data)
  }
}